### DISCUSSION TITLE
after updating Firefly III - 500 Internal Server Error :(

### DISCUSSION THREAD

**Discussion by lepidas** [firefly-iii/firefly-iii#8988] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hi to all, I ran Firefly for the last 3 years but now updating from 6.1.15 to 6.1.18 everything went wrong. I tried everything all solutions I could understand but nothing. Now I ask for your help. Last thing I did was to run again 6.1.15

### Debug information

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.15 / v2.0.14 / 24 (exp. 24)</td></tr><tr><td>PHP version</td><td>8.3.8 (64bits) / fpm-fcgi / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>2097152 (2 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, </td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Amsterdam + Europe/Athens</td></tr><tr><td>App environment</td><td>local, debug: true</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td></td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has piggy banks">:pig:</span> <span title="Has rules">:wrench:</span></td></tr><tr><td>Session start</td><td>2024-01-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-12-31 23:59:59</td></tr><tr><td>View range</td><td>1Y</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:126.0) Gecko/20100101 Firefox/126.0</td></tr></tbody></table>

### Additional info

 An exception has been thrown during the rendering of a template ("file_put_contents(/var/www/html/firefly-iii/storage/framework/cache/data/1b/e5/1be51c18f6cad86450649f1668f9e67c79db1038): Failed to open stream: No such file or directory").

**Comment by JC5** [comment, marked as answer]
https://docs.firefly-iii.org/references/faq/install/#unable-to-write-to-cache-directory

**Reply by lepidas** [reply]
That did it. Thank you very much for your time
